tensorflow/cc/tools/freeze_saved_model.cc

```diff
@@ -204,16 +227,35 @@ Status FreezeGraphDef(const SavedModelBundle& saved_model_bundle,
     if (variable_node_names.find(node.name()) != variable_node_names.end()) {
       ConvertVariableToConstant(node, variable_to_value_map[node.name()],
                                 frozen_graph_def->add_node());
-    } else if (node.op() == "ReadVariableOp" &&
-               variable_node_names.find(node.input(0)) !=
-                   variable_node_names.end()) {
-      // If the node is a ReadVariableOp, its input VarHandleOp will be
-      // converted to a Constant, so we will need to convert it to an Identity.
-      ConvertReadVariableOpToIdentity(node, frozen_graph_def->add_node());
-    } else {
-      // If the node isn't a variable, just copy the node as-is.
-      *frozen_graph_def->add_node() = node;
+      continue;
+    } else if (node.op() == "ReadVariableOp" || node.op() == "Identity") {
```

Comment:
I don't understand why "|| node.op() == "Identity"" and the "else" in line 244 is necessary.
Can you explain?
Does the test that you added test this situation?